hazelcast/src/main/java/com/hazelcast/internal/monitors/PerformanceMonitor.java

```diff
@@ -16,61 +16,107 @@
- * point in time.
+ * The PerformanceMonitor is a tool that provides insights in internal metrics. Currently the content of the
+ * {@link MetricsRegistry} is being dumped.
  */
-public class PerformanceMonitor extends Thread {
+public class PerformanceMonitor {
 
-    private final ILogger logger;
+    final MetricsRegistry metricRegistry;
+    final HazelcastInstanceImpl hazelcastInstance;
+    final ILogger logger;
+    final InternalOperationService operationService;
+    final PerformanceLogFile performanceLogFile;
     private final Node node;
-    private final int delaySeconds;
-    private final InternalOperationService operationService;
-    private final ConnectionManager connectionManager;
-
-    public PerformanceMonitor(HazelcastInstanceImpl hazelcastInstance, int delaySeconds) {
-        super(hazelcastInstance.node.getHazelcastThreadGroup().getInternalThreadGroup(),
-                hazelcastInstance.node.getHazelcastThreadGroup().getThreadNamePrefix("PerformanceMonitor"));
-        setDaemon(true);
+    private final MonitorThread monitorThread;
+    private final boolean enabled;
 
-        this.delaySeconds = delaySeconds;
+    public PerformanceMonitor(HazelcastInstanceImpl hazelcastInstance) {
+        this.hazelcastInstance = hazelcastInstance;
         this.node = hazelcastInstance.node;
-        this.logger = node.getLogger(PerformanceMonitor.class.getName());
         this.operationService = node.nodeEngine.getOperationService();
-        this.connectionManager = node.connectionManager;
+        this.logger = node.getLogger(PerformanceMonitor.class);
+        this.metricRegistry = hazelcastInstance.node.nodeEngine.getMetricsRegistry();
+        this.enabled = node.getGroupProperties().PERFORMANCE_MONITOR_ENABLED.getBoolean();
+        this.performanceLogFile = new PerformanceLogFile(this);
+        this.monitorThread = initMonitorThread();
+    }
+
+    private MonitorThread initMonitorThread() {
+        if (!enabled) {
+            return null;
+        }
+
+        GroupProperties props = node.getGroupProperties();
+        int delaySeconds = props.PERFORMANCE_MONITOR_DELAY_SECONDS.getInteger();
+
+        HazelcastThreadGroup threadGroup = node.getHazelcastThreadGroup();
+        return new MonitorThread(threadGroup, delaySeconds);
     }
 
-    @Override
-    public void run() {
-        StringBuffer sb = new StringBuffer();
+    public PerformanceMonitor start() {
+        if (!enabled) {
+            logger.finest("PerformanceMonitor disabled");
+            return this;
+        }
 
-        while (node.isActive()) {
-            sb.append("\n");
-            sb.append("ConnectionManager metrics\n");
-            connectionManager.dumpPerformanceMetrics(sb);
-            sb.append("OperationService metrics\n");
-            operationService.dumpPerformanceMetrics(sb);
+        logger.info("PerformanceMonitor started");
```

Comment:
this feels spammy to me too - we don't log every starting component on INFO level.